packages/react/src/analytics/integrations/GA4/eventMapping.ts

```diff
@@ -584,18 +626,20 @@ const getShareParametersFromEvent = eventProperties => ({
 /**
  * Returns the signup newsletter parameters formatted for the GA4 event.
  *
- * @param {object} eventProperties - Properties from a track event.
+ * @param eventProperties - Properties from a track event.
  *
- * @returns {object} Parameters formatted for the GA4's sign_up_newsletter event.
+ * @returns Parameters formatted for the GA4's sign_up_newsletter event.
  */
-const getSignupNewsletterParametersFromEvent = eventProperties => {
+const getSignupNewsletterParametersFromEvent = (
```

Comment:
This can be more type-safe with this:

```js
const getSignupNewsletterParametersFromEvent = (
  eventProperties: EventProperties,
) => {
  type GenderValues = keyof typeof SignupNewsletterGenderMappings;
  type GenderObject = { id: GenderValues; name?: string };

  const genderArray: Array<string> = (
    Array.isArray(eventProperties.gender)
      ? eventProperties.gender
      : new Array(eventProperties.gender)
  ).map((gender: GenderValues | GenderObject) => {
    return (
      (gender as GenderObject).name ||
      SignupNewsletterGenderMappings[(gender as GenderObject).id ?? gender]
    );
  });
```